ClaudiaRojasSoto/propertiesManagment
# Feature/initial configuration

Title: Initial Setup and Linters Configuration

Body:
## Summary
This pull request includes the initial setup for our Rails + React application and the configuration of essential linters to ensure code quality and consistency across the project.

## Changes
- Set up the Rails application with React support using Webpacker.
- Configured RuboCop for Ruby code style enforcement, with a customized `.rubocop.yml` file to meet our project's coding standards.
- Integrated ESLint for linting our JavaScript and JSX code, ensuring adherence to best practices and our specific style guides. The ESLint configuration is defined in the `.eslintrc.json` file.
- Added stylelint to manage and enforce styling rules for our CSS and SCSS files. The configuration for stylelint is provided in the `.stylelintrc.json` file.
- All linter configurations have been added to ensure they align with our project's coding conventions and best practices.

## Purpose
The purpose of this setup is to lay a solid foundation for our application's development, ensuring that all code written is clean, consistent, and error-free. By integrating these linters into our workflow, we aim to improve the maintainability and quality of our codebase, making it easier for developers to contribute and collaborate on this project.

## How to Test
- Ensure you have the latest version of Node.js, Ruby, and Rails installed.
- Pull this branch into your local environment.
- Run `bundle install` and `yarn install` to install the required dependencies for Rails and React, respectively.
- Execute `bundle exec rubocop`, `npx eslint .`, and `npx stylelint "**/*.scss"` to test the linters' configurations. There should be no offenses detected.
- Start the Rails server with `rails s` and visit `http://localhost:3000` to ensure the application runs without errors.

Please review the changes and configurations. Your feedback is highly appreciated to ensure we are setting up our project's foundation correctly.

Thank you!
